pkg/cloud/instance.go

```diff
@@ -268,21 +270,19 @@ func (c *client) GetOrCreateVMInstance(
 	if err != nil {
 		c.customMetrics.EvaluateErrorAndIncrementAcsReconciliationErrorCounter(err)
 
-		// Just because an error was returned doesn't mean a (failed) VM wasn't created and will need to be dealt with.
-		// Regretfully the deployVMResp may be nil, so we need to get the VM ID with a separate query, so we
-		// can return it to the caller, so they can clean it up.
-		listVirtualMachineParams := c.cs.VirtualMachine.NewListVirtualMachinesParams()
-		listVirtualMachineParams.SetTemplateid(templateID)
-		listVirtualMachineParams.SetZoneid(fd.Spec.Zone.ID)
-		listVirtualMachineParams.SetNetworkid(fd.Spec.Zone.Network.ID)
-		listVirtualMachineParams.SetName(csMachine.Name)
-		listVirtualMachinesResponse, err2 := c.cs.VirtualMachine.ListVirtualMachines(listVirtualMachineParams)
-		if err2 != nil || listVirtualMachinesResponse.Count <= 0 {
-			c.customMetrics.EvaluateErrorAndIncrementAcsReconciliationErrorCounter(err2)
-			return err
+		// CloudStack may have created the VM even though it reported an error. We attempt to
```

Comment:
I refactored this because the `GetOrCreateVMInstance()` method had a high cyclomatic complexity.